functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
+        try:
+            assert 'title' in task_req_body, "New task request body did not contain field: 'title'"
+            assert 'description' in task_req_body, "New task request body did not contain field: 'description'"
+            assert 'dueDate' in task_req_body, "New user request body did not contain field: 'dueDate'"
+        except AssertionError as task_req_body_content_error:
+            logging.critical('New user request body did not contain fields to update a task')
+            return func.HttpResponse(task_req_body_content_error.args[0], status_code=400)
+        logging.info('New user request body contained necessary fields to update a task')  
+        # Unpack task data
+        title = task_req_body.get('title')
+        description = task_req_body.get('description')
+        dueDate = datetime.strptime(task_req_body.get('dueDate'), '%d/%m/%y %H:%M:%S')     
+        # update task: title and description
+        sql_query = """UPDATE [dbo].[tasks]
+        SET title = '{}', description = '{}', dueDate = '{}'
+        WHERE userId = ? AND taskId = ?""".format(title, description, dueDate)
+        try:
+            rowcount = cursor.execute(sql_query, param1, param2).rowcount
+            if not rowcount:
+                logging.error('No record with the requested parameters exists in db')
+                return func.HttpResponse('No record to update', status_code=500)                
+            logging.debug('UPDATE query executed')
+            logging.info(f"Executed the query: {rowcount} rows affected for taskId {param2}")
+            return func.HttpResponse(status_code=200)
+        except Exception as e:
+            logging.critical('Unable to execute the query')
+            return func.HttpResponse("Error: %s" % str(e), status_code=400)
+    finally:
+        #commits changes to db
         cnxn.commit()
-        return rowcount
-#irrespective of results, closes connection
-    finally: 
+        #properly closes the connection
         cursor.close()
         cnxn.close()
-        logging.info('Closed the connection')
+        logging.info('Closed the db connection')    
 
 #DELETE API method function
-def delete(param):
-    return ('You selected POST method with {param} values. Functionality under construction')
-            # STARTER CODE FOR NEXT SPRINT, SQL QUERY TESTED
-            #     # #delete row, client passes in userId and taskId
-            #     sql_query = ("""DELETE FROM [dbo].[tasks] WHERE userId = ? AND taskId = ?""")
-            #     cursor.execute(sql_query, userId, taskId)
-            #     logging.info('Executed the query')
-            #     cnxn.commit()
+def delete(param1, param2):
+    #connects to db
+    cnxn = connect()
+    cursor = cnxn.cursor()
+    try:
+        logging.info('opened connection')        
+        logging.info('Going to execute DELETE task query')
+        # # task_req_body input check

```

Comment:
you may want to consider deleting these comments. I think these are leftovers from the update function. lines 94 - 107

Reply:
good call